internal/controllers/utils/utils.go

```diff
@@ -33,9 +34,12 @@ func MatchesIngressClass(obj client.Object, controllerIngressClass string, isDef
 			return true
 		}
 	}
-	if objectIngressClass == controllerIngressClass || objectKnativeClass == controllerIngressClass {
+
+	switch controllerIngressClass {
+	case objectIngressClass, objectKnativeClass, objectKnativeClassAlt:
```

Comment:
I agree that it would be beneficial to split but perhaps only in case where we'd have tests that test each of annotations and/or the ingressClass field.

WDYT @mflendrich ?

Reply:
technically speaking, we do (I think). And if we don't, then that's a surprise that requires action for sake of QA.

Reply:
Agreed, let's change that.

Reply:
Done